In [1]:
import datetime
import calendar

def weekdays_count(start,end):
    Starting_Date = datetime.datetime.strptime(start, '%d/%m/%Y')
    Ending_Date = datetime.datetime.strptime(end, '%d/%m/%Y')
    weeks = {}
    for i in range((Ending_Date - Starting_Date).days):
        day = calendar.day_name[(Starting_Date + datetime.timedelta(days = i+1)).weekday()]
        weeks[day] = weeks[day] + 1 if day in weeks else 1
    return weeks

In [2]:
print weekdays_count("1/1/1990", "31/12/2000")

{'Monday': 573, 'Tuesday': 574, 'Friday': 574, 'Wednesday': 574, 'Thursday': 574, 'Sunday': 574, 'Saturday': 574}


So, the total number of Thursdays between 1990 & 2000 are 574.